### DISCUSSION TITLE
Private library per user and one shared to all

### DISCUSSION THREAD

**Discussion by Samega7Cattac** [photoprism/photoprism#4233] [Q&A] [answered]
Hi,
I'm trying to setup a instance where each user has a private library (so other users can't see) that corresponds to originals folder.

So `user1` has access to the library `user1`
And `user2` has access to the library `user2`
But then both `user1` and `user2` has access to the library `shared`

I have setup the accounts already but I'm unable to find a way to limit the access to the libraries.
From what I noticed I need to be a paid user so I got the Essential subscription.

**Comment by graciousgrey** [comment, marked as answer]
Hi,

Our user roles are intended for situations where you want other people to have access to your library, such as giving family members access to your pictures without granting write permissions or exposing private content.

It is recommended to set up additional instances if you have multiple users in a family, so that everyone can manage their own files independently. Implementing a more convenient library manager is planned for the future :)

https://docs.photoprism.app/user-guide/users/libraries/

**Reply by mniami** [reply]
Hi,
what do you mean by that, having like multiple services installed per user?
That seems to be a bit weird to be honest.

**Reply by lastzero** [reply]
If you just want to index multiple, different directories then you already do so by mounting the corresponding folders. You do not need to run multiple instances (or databases) for this.

**Reply by mniami** [reply]
Thanks @lastzero, not fully understand. Indexing multiple directories doesn't allow to have private folders/libraries per user or I'm wrong (mounting them too).
I just realized that photoprism had a bit different target, but the app has been written so efficiently that I would love to see that feature to allow my family to have theirs private photos really private. 
Currently we are using Nextcloud, but it is not that efficient as Photoprism, which was written in Go.
If you have any links how to set it up I would be appreciate, cheers!
